flybywiresim/a32nx
# feat(efb): flyPad battery

Fixes #5615

## Summary of Changes
Manage battery level for EFB fly pad.

Battery is charged as long as DC2 bus is powerd. If not, battery discharges for 240min using a very simple linear function. Charging takes 180min from empty to full - again using a linear function model.

In case the battery is empty, flypad shuts down completly and needs to be charaged again to be functional again. 

## Screenshots (if necessary)

### (1) Batttery full, DC2 bus powered
![01_EfbBattery_Full-charged](https://user-images.githubusercontent.com/768655/146768861-6fba6c58-6c01-4591-b76b-7358b5b589dc.png)

### (2) Battery discharged, DC2 bus not powered
![02_EfbBattery-discharge](https://user-images.githubusercontent.com/768655/146769049-b52910bc-4548-416b-9a7d-5a38e9779eca.png)

### (3) Battery at warning level
![03_EfbBattery-warning-discharge](https://user-images.githubusercontent.com/768655/146769088-7f6694c2-0764-4c3b-bdb6-d8a0edb8206b.png)

### (4) Battery empty, flypad not operational
![04_EfbBattery-empty](https://user-images.githubusercontent.com/768655/146769139-0ba5d95e-6462-45da-892e-739220a6a883.png)

### (5) Flypad not yet operational (close to empty) but charged
![05_EfbBattery-empty-charge](https://user-images.githubusercontent.com/768655/146769234-e4c6d701-530e-442b-8a1b-fc23d0033802.png)

### (6) Flypad operational, still being charged
![06_EfbBattery-charge](https://user-images.githubusercontent.com/768655/146769303-5f936e20-77e0-41bb-8764-1e0c1bb54069.png)

## References
n/a

Battery life parameters taken from own experience but not from a real aircraft ;-).

## Additional context

The battery model is based on ABSOLUTE TIME, therefore changing the time in the sim should also change the battery level accordingly. 

Discord username (if different from GitHub): `geholger [Z+1]#9751`

## Testing instructions

1. Start EFB
2. Disconnect DC2 bus --> fly pad should discharge and battery level / icon should change over time (see (1))
3. Connect DC2 bus --> fly pad should discharge (see (6), battery level should increase
4. Disconect DC2 bus and wait until battey is empty (you might want to change the sim time)
--> at <8%, battery warning should be indicated (see (3))
--> at 0%,  battery level, fly pad should shut down (see (4))
5. Connect DC2 bus
--> Fly pad should remain inoperative for a while but "charging" should be shown (see (5))
--> After a while, fly pad should start up and be functional again (see (6))

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
